openapi2beans/pkg/generator/yaml2schematypes.go

```diff
@@ -0,0 +1,282 @@
+/*
+ * Copyright contributors to the Galasa project
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package generator
+
+import (
+	"log"
+	"strings"
+
+	openapi2beans_errors "github.com/dev-galasa/buildutils/openapi2beans/pkg/errors"
+	"gopkg.in/yaml.v3"
+)
+
+const (
+	OPENAPI_YAML_SCHEMAS_PATH        = "#/components/schemas"
+	OPENAPI_YAML_KEYWORD_COMPONENTS  = "components"
+	OPENAPI_YAML_KEYWORD_SCHEMAS     = "schemas"
+	OPENAPI_YAML_KEYWORD_DESCRIPTION = "description"
+	OPENAPI_YAML_KEYWORD_PROPERTIES  = "properties"
+	OPENAPI_YAML_KEYWORD_TYPE        = "type"
+	OPENAPI_YAML_KEYWORD_REQUIRED    = "required"
+	OPENAPI_YAML_KEYWORD_ITEMS       = "items"
+	OPENAPI_YAML_KEYWORD_ALLOF       = "allOf"
+	OPENAPI_YAML_KEYWORD_ONEOF       = "oneOf"
+	OPENAPI_YAML_KEYWORD_ANYOF       = "anyOf"
+	OPENAPI_YAML_KEYWORD_REF         = "$ref"
+	OPENAPI_YAML_KEYWORD_ENUM        = "enum"
+)
+
+var arrayDimensions = 0
```

Comment:
What does this control ? It's not obvious ? which array ? Which dimension ?

Reply:
added comments

Reply:
added comments